config/tls.go

```diff
@@ -101,6 +105,37 @@ func (c *TLSConfig) Finalize() {
 			StringPresent(c.ServerName) ||
 			BoolPresent(c.Verify))
 	}
+	if c.CACert == nil {
```

Comment:
Just wondering, are we using the non-Consul/non-Vault specific `Finalize` anywhere else?